vra/resource_storage_profile_vsphere_test.go

```diff
@@ -0,0 +1,100 @@
+package vra
+
+import (
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+	"github.com/vmware/vra-sdk-go/pkg/client/cloud_account"
+	"github.com/vmware/vra-sdk-go/pkg/client/storage_profile"
+)
+
+func TestAccVRAStorageProfileVsphereBasic(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheckStorageProfileVsphere(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckVRAStorageProfileVsphereDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCheckVRAStorageProfileVsphereConfig(),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					testAccCheckVRAStorageProfileVsphereExists("vra_storage_profile_vsphere.my-storage-profile-vsphere"),
+					resource.TestCheckResourceAttr(
+						"vra_storage_profile_vsphere.my-storage-profile-vsphere", "name", "vra-storage-profile-vsphere-test-0"),
+					resource.TestCheckResourceAttr(
+						"vra_storage_profile_vsphere.my-storage-profile-vsphere", "description", "my storage profile vsphere"),
+					resource.TestCheckResourceAttr(
+						"vra_storage_profile_vsphere.my-storage-profile-vsphere", "default_item", "true"),
+					// TODO: Enable after https://jira.eng.vmware.com/browse/VCOM-20943 is resolved
+					//resource.TestCheckResourceAttr(
+					//	"vra_storage_profile_vsphere.my-storage-profile-vsphere", "disk_mode", "independent-persistent"),
+					resource.TestCheckResourceAttr(
+						"vra_storage_profile_vsphere.my-storage-profile-vsphere", "limit_iops", "1000"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckVRAStorageProfileVsphereExists(n string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return fmt.Errorf("not found: %s", n)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("no storage profile vsphere ID is set")
+		}
+
+		return nil
+	}
+}
+
+func testAccCheckVRAStorageProfileVsphereDestroy(s *terraform.State) error {
+	apiClient := testAccProviderVRA.Meta().(*Client).apiClient
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type == "vra_cloud_account_vsphere" {
+			_, err := apiClient.CloudAccount.GetVSphereCloudAccount(cloud_account.NewGetVSphereCloudAccountParams().WithID(rs.Primary.ID))
+			if err == nil {
+				return fmt.Errorf("Resource 'vra_cloud_account_vsphere' still exists with id %s", rs.Primary.ID)
+			}
+		}
```

Comment:
This is not needed as the config only has data source for vra_cloud_account_vsphere, not reource.